src/web/middleware.rs

```diff
@@ -128,6 +129,7 @@ impl Middleware<ServerState> for DbTransaction {
                     req.extensions_mut().insert((db, in_transaction));
                     future::ok(None)
                 })
+                .map_err(Into::into)
```

Comment:
this doesn't look needed (and we could use less of them oh god they're everywhere)

Reply:
Honestly, I'm still not sure what's eating the error. That was more a hail Mary. Now that the clippy stuff is in place, I'll refactor and strip a lot of these out. 

My hunch is whatever is processing the future is eating the error being generated in the `future:err()` and we might be missing a handler at a higher level. I'm still trying to get my head around the futures stuff (my apologies for being dense), that's why this is still a WIP.